src/dict.c

```diff
@@ -778,6 +788,11 @@ dictEntry *dictFindByHash(dict *d, const void *key, const uint64_t hash) {
     /* Rehash the hash table if needed */
     _dictRehashStepIfNeeded(d,idx);
 
+    /* Check if we can use the compare function with length to avoid recomputing length of key always */
+    keyCmpFuncWithLen cmpFuncWithLen = dictGetKeyCmpFuncWithLen(d);
+    keyLenFunc keyLenFunc = dictGetKeyLenFunc(d);
```

Comment:
```suggestion
    keyCmpFuncWithLen cmpFuncWithLen = d->type->keyCompareWithLen;
    keyLenFunc keyLenFunc = d->type->keyLen;
```

then we can eliminate dictGetKeyCmpFuncWithLen() and dictGetKeyLenFunc(),

Reply:
agree.
Needed to move keyLenFunc and keyCmpFuncWithLen signatures to dict.h.
All addressed in https://github.com/redis/redis/pull/13792/commits/99b60fef718aa9e4edb804217ab66ed4c19f56cd